### PR TITLE
scripts: add verify links check and fix some broken links

### PR DESCRIPTION
**Pull request by cofyc**
### What is changed, added or deleted? 

Add a script to verify internal links and fix some broken links under dev and v3.0 directories.

Why: Avoid broken links in CI phrase.

### Which version does your change affect? 

dev
v3.0

### Checklist 

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

File: scripts/markdown-link-check.tpl [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "ignorePatterns": [
+        {
+            "pattern": "^(http|https|ftp|mailto):"
+        },
+        {
+            "pattern": "\\.\\./media/"
```
Comment (cofyc) [inline scripts/markdown-link-check.tpl]: this case is hard to cover, so I ignore it for now (we can cover it later)

File: scripts/markdown-link-check.tpl [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+    "ignorePatterns": [
+        {
+            "pattern": "^(http|https|ftp|mailto):"
```
Comment (cofyc) [inline scripts/markdown-link-check.tpl]: ignore all external links and mailto links

File: scripts/markdown-link-check.tpl [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            "pattern": "^(?!(\\.|/media/))",
+            "replacement": "<DOC_ROOT>/"
+        },
+        {
+            "pattern": "^/media/",
+            "replacement": "<ROOT>/media/"
+        },
+        {
+            "pattern": "#.+",
+            "replacement": ""
```
Comment (cofyc) [inline scripts/markdown-link-check.tpl]: anchors are ignored for now

File: scripts/markdown-link-check.tpl [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        }
+    ],
+    "replacementPatterns": [
+        {
+            "pattern": "^(?!(\\.|/media/))",
+            "replacement": "<DOC_ROOT>/"
+        },
+        {
+            "pattern": "^/media/",
+            "replacement": "<ROOT>/media/"
```
Comment (cofyc) [inline scripts/markdown-link-check.tpl]: `/media/...` references media source in the root of the repository.

File: scripts/verify-links.sh [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+if ! which markdown-link-check &>/dev/null; then
+    sudo npm install -g markdown-link-check@3.7.3
+fi
+
+CONFIG_TMP=$(mktemp)
+
+trap 'rm -f $CONFIG_TMP' EXIT
+
+# TODO fix later
+IGNORE_DIRS=(v1.0 v2.0 v2.1 v2.1-legacy)
```
Comment (cofyc) [inline scripts/verify-links.sh]: there are some broken links in pre-v3.0 directories, but I don't want to make this PR too complicated. If this script is accepted, we can fix it later.

File: v3.0/reference/sql/statements/load-data.md [hunk 6]
```diff
@@ -49,4 +49,3 @@ Records: 815264  Deleted: 0  Skipped: 0  Warnings: 0
 
 * [INSERT](/reference/sql/statements/insert.md)
 * [Transaction Model](/reference/transactions/transaction-model.md)
-* [Import Example Database](/how-to/get-started/import-example-database.md)
```
Comment (cofyc) [inline v3.0/reference/sql/statements/load-data.md]: I can't find where it is. Was it deleted already?

Comment (lilin90) [reply v3.0/reference/sql/statements/load-data.md]: @cofyc Yes. We only keep this doc in the English version. Just delete this line.

File: v3.0/how-to/deploy/orchestrated/ansible.md [hunk 7]
```diff
@@ -500,7 +500,7 @@ TiKV1-1 ansible_host=172.16.10.4 deploy_dir=/data1/deploy
 | cluster_name | 集群名称，可调整 |
 | tidb_version | TiDB 版本，TiDB-Ansible 各分支默认已配置 |
 | process_supervision | 进程监管方式，默认为 systemd，可选 supervise |
-| timezone | 新安装 TiDB 集群第一次启动 bootstrap（初始化）时，将 TiDB 全局默认时区设置为该值。TiDB 使用的时区后续可通过 `time_zone` 全局变量和 session 变量来修改，参考[时区支持](../sql/time-zone.md)。 默认为 `Asia/Shanghai`，可选值参考 [timzone 列表](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)。 |
+| timezone | 新安装 TiDB 集群第一次启动 bootstrap（初始化）时，将 TiDB 全局默认时区设置为该值。TiDB 使用的时区后续可通过 `time_zone` 全局变量和 session 变量来修改，参考[时区支持](how-to/configure/time-zone.md)。 默认为 `Asia/Shanghai`，可选值参考 [timzone 列表](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)。 |
```
Comment (xiaojingchen) [inline v3.0/how-to/deploy/orchestrated/ansible.md]:
```suggestion
| timezone | 新安装 TiDB 集群第一次启动 bootstrap（初始化）时，将 TiDB 全局默认时区设置为该值。TiDB 使用的时区后续可通过 `time_zone` 全局变量和 session 变量来修改，参考[时区支持](/how-to/configure/time-zone.md)。 默认为 `Asia/Shanghai`，可选值参考 [timzone 列表](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)。 |
```

File: v3.0/how-to/deploy/tidb-in-kubernetes/aws-eks.md [hunk 8]
```diff
@@ -347,7 +347,7 @@ terraform destroy
 
 上述文档中介绍的 Terraform 脚本组合了多个 Terraform 模块：
 
-- `tidb-operator` 模块，用于创建 EKS 集群并在 EKS 集群上安装配置 [TiDB Operator](how-to/deploy/tidb-operator.md)。
+- `tidb-operator` 模块，用于创建 EKS 集群并在 EKS 集群上安装配置 [TiDB Operator](how-to/deploy/tidb-in-kubernetes/tidb-operator.md)。
```
Comment (xiaojingchen) [inline v3.0/how-to/deploy/tidb-in-kubernetes/aws-eks.md]:
```suggestion
- `tidb-operator` 模块，用于创建 EKS 集群并在 EKS 集群上安装配置 [TiDB Operator](/how-to/deploy/tidb-in-kubernetes/tidb-operator.md)。
```

File: dev/faq/tidb.md [hunk 9]
```diff
@@ -39,7 +39,7 @@ TiDB 目前还不支持触发器、存储过程、自定义函数、外键，除
 
 #### 1.1.7 TiDB 具备高可用的特性吗？
 
-TiDB 天然具备高可用特性，TiDB、TiKV、PD 这三个组件都能容忍部分实例失效，不影响整个集群的可用性。具体见 [TiDB 高可用性](/key-features/#高可用)。
+TiDB 天然具备高可用特性，TiDB、TiKV、PD 这三个组件都能容忍部分实例失效，不影响整个集群的可用性。具体见 [TiDB 高可用性](/key-features.md#高可用)。
```
Comment (tennix) [inline dev/faq/tidb.md]: I think the markdown file will be converted to a link without suffix `.md`.

Comment (cofyc) [reply dev/faq/tidb.md]: not for now, it's converted to `https://pingcap.com/key-features/#%E9%AB%98%E5%8F%AF%E7%94%A8` which is not right

Comment (lilin90) [reply dev/faq/tidb.md]: > I think the markdown file will be converted to a link without suffix `.md`.

Yes. The link on the website does not contain `.md`, while in Markdown file, we need the `.md`.

**Comment by cofyc** [comment]
ah, I found a problem. Because `alias` are not covered, when a file is moved, all references are required to be updated for now.

When a file was moved, all other references are required to be updated for now, even if alias are given
 12 #   - TODO relaxing this requirement?

**Comment by cofyc** [comment]
Do you think is it a problem? IMO, it's better to update all old references on our website.

**Comment by lilin90** [comment]
@cofyc Currently, we suggest updating all reference links to the latest link. In addition, if some words are linked to a 2nd level heading in another file, only using the latest link can support jumping to the exact position.

**Comment by cofyc** [comment]
Then, I think it's ok to merge this and check if it's going well in our workflow.

**Comment by lilin90** [comment]
@cofyc Would you please also help add this check to pingcap/docs if it works well? We'd appreciate it a lot. ^_^

**Comment by cofyc** [comment]
sure, I'm working on it.

**Pull request merged**